### PR TITLE
Added Razorpay Integration Backend

### PR DESCRIPTION
**Pull request by rohitj921**
- added  payment model
- added payment routes to create order , verify payment signature , storing and fetching details from  the database
- added payment controllers to manage routes logic
- added services for Razorpay integration
- added "razorpay" module dependancy 
- added  payment routes to index.js

### CODE REVIEW

File: backend/controllers/Payment/ordersController.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+const Payment = require('../../models/Payment/paymentModel')
+
+
+const newOrder = async (req, res) => {
+    try {
+        const { amount, currency,clientName,clientId, therapistName,therapistId} = req.body;
+        const options = {
+            amount: amount * 100,
+            currency: currency,
+            // receipt: crypto.randomBytes(16).toString('hex'),
```
Comment (chandan24042001s) [inline backend/controllers/Payment/ordersController.js:12]: Remove comment if it has no use in future

File: backend/controllers/Payment/paymentDetails.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+const Payment = require('../../models/Payment/paymentModel')
+
+
+// Get all Payment records
+
+const getAllPayments = async (req, res) => {
+    try {
+        const payments = await Payment.find();
```
Comment (chandan24042001s) [inline backend/controllers/Payment/paymentDetails.js:8]: .find({}), Maybe for finding in mongodb. It should be used. Please verify this

File: backend/controllers/Payment/ordersController.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+const razorpay = require('../../services/razorpayService');
+const crypto = require('crypto');
```
[the remark this answers is not in the document]
Comment (rohitj921) [reply backend/controllers/Payment/ordersController.js:2]: That was to generate payment receipt if it is required in frontend `receipt: crypto.randomBytes(16).toString('hex')`.
I'll remove both module import and receipt for now.

**Review by chandan042001s — APPROVED** [review]

File: backend/controllers/Payment/paymentDetails.js (hunk 2 above)
Comment (rohitj921) [reply backend/controllers/Payment/paymentDetails.js]: yeah I've changed that along with other comments

File: backend/controllers/Payment/ordersController.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rohitj921) [reply backend/controllers/Payment/ordersController.js]: I forgot to mention  it is required to verify payment signature so couldn't remove.

**Pull request merged**